src/helpers/PromiseCollector.ts

```diff
@@ -0,0 +1,25 @@
+export default class PromiseCollector {
+  private _promises: Array<Promise<any>> = [];
```

Comment:
```suggestion
  public final promises: Array<Promise<any>> = [];
```

Similarly, it seems like this would obviate the need for a helper. However, I _do_ like that the current method exposes a line that can be covered in codecov (to see if the public field is ever read)

Reply:
i did the `get` helper to convert the type to `ReadonlyArray` otherwise the readonly var just means you can't overwrite it but you can still call like `array.push` or whatever to modify it.

Reply:
oooh, interesting. okies. I'm still not used to using getters in JS.

Reply:
it would be nice if typescript had a way to designate the type as like `PrivateArray` or something so i didn't have to jigger the type to be different internally vs externally using that getter. i wonder if they have that...